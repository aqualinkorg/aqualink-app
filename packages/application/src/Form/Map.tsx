import React from "react";
import { Map, TileLayer, Marker } from "react-leaflet";
import L from "leaflet";
import { withStyles, WithStyles, createStyles } from "@material-ui/core";

import marker from "../assets/marker.png";

const myIcon = L.icon({
  iconUrl: marker,
  iconSize: [20, 30],
  iconAnchor: [10, 30],
  popupAnchor: [0, -41],
});

const ReefMap = ({ center, classes }: ReefMapProps) => {
  return (
    <Map className={classes.map} center={center} zoom={8}>
      <TileLayer url="https://server.arcgisonline.com/ArcGIS/rest/services/World_Imagery/MapServer/tile/{z}/{y}/{x}" />
      <Marker icon={myIcon} position={center} />
    </Map>
  );
};

const styles = () => {
  return createStyles({
    map: {
      height: "100%",
      width: "100%",
      borderRadius: 4,
    },
  });
};

interface ReefMapIncomingProps {
  center: [number, number];
}

type ReefMapProps = WithStyles<typeof styles> & ReefMapIncomingProps;

export default withStyles(styles)(ReefMap);
