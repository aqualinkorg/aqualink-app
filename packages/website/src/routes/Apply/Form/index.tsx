import React, { useCallback, useState, useEffect } from "react";
import {
  withStyles,
  WithStyles,
  createStyles,
  Theme,
  Typography,
  TextField,
  Button,
} from "@material-ui/core";
import { useForm } from "react-hook-form";
import {
  KeyboardDatePicker,
  MuiPickersUtilsProvider,
} from "@material-ui/pickers";
import DateFnsUtils from "@date-io/date-fns";
import moment from "moment";
import { ReefApplication, ReefApplyParams } from "../../../store/Reefs/types";

const Form = ({
  reefName,
  application,
  agreed,
  handleFormSubmit,
  classes,
}: FormProps) => {
  const [installationSchedule, setInstallationSchedule] = useState<Date | null>(
    null
  );

  useEffect(() => {
    if (application?.installationSchedule) {
      setInstallationSchedule(new Date(application.installationSchedule));
    }
  }, [application]);

  const { register, errors, handleSubmit } = useForm({
    reValidateMode: "onSubmit",
  });

  const handleInstalationChange = (date: Date | null) => {
    if (date) {
      setInstallationSchedule(date);
    }
  };

  const formSubmit = useCallback(
    (data: any) => {
      const params: ReefApplyParams = {
        fundingSource: data.fundingSource,
        permitRequirements: data.permitRequirements,
        installationResources: data.installationResources,
        installationSchedule: new Date(data.installationSchedule).toISOString(),
      };
      handleFormSubmit(data.siteName, params);
    },
    [handleFormSubmit]
  );

  return (
    <form className={classes.form} onSubmit={handleSubmit(formSubmit)}>
      <Typography className={classes.formTitle} variant="h3">
        Your Site
      </Typography>

      <TextField
        className={`${classes.formField} ${classes.textField}`}
        variant="outlined"
        inputProps={{ className: classes.textField }}
        fullWidth
        placeholder="Reef Name e.g. 'Sombrero Reef'"
        disabled={Boolean(reefName)}
        defaultValue={reefName || null}
        name="siteName"
        inputRef={register({
          required: "This is a required field",
        })}
        error={!!errors.siteName}
        helperText={errors?.siteName?.message || ""}
      />

      <Typography className={classes.additionalInfo}>
        Please provide some additional information for each reef:
      </Typography>

      <Typography>Permitting</Typography>
      <TextField
        className={`${classes.formField} ${classes.textField}`}
        variant="outlined"
        inputProps={{ className: classes.textField }}
        fullWidth
        multiline
        rows={2}
        defaultValue={application?.permitRequirements || null}
        placeholder="Please describe the permitting requirements. Please be sure to mention the authority having jurisdiction"
        name="permitRequirements"
        inputRef={register({
          required: "This is a required field",
        })}
        error={!!errors.permitRequirements}
        helperText={errors?.permitRequirements?.message || ""}
      />

      <Typography>Funding Source</Typography>
      <TextField
        className={`${classes.formField} ${classes.textField}`}
        variant="outlined"
        inputProps={{ className: classes.textField }}
        fullWidth
        multiline
        rows={2}
        defaultValue={application?.fundingSource || null}
        placeholder="Funding source for import duties and shipping. Please describe the funding source for the import duties and shipping costs"
        name="fundingSource"
        inputRef={register({
          required: "This is a required field",
        })}
        error={!!errors.fundingSource}
        helperText={errors?.fundingSource?.message || ""}
      />

      <Typography>Schedule for installation</Typography>
      <Typography className={classes.scheduleDescription}>
        What is the soonest date you could install the Smart Buoy and conduct a
        survey?
      </Typography>
      <MuiPickersUtilsProvider utils={DateFnsUtils}>
        <KeyboardDatePicker
          className={classes.formField}
          disableToolbar
          format="MM/dd/yyyy"
          id="installationSchedule"
          name="installationSchedule"
          autoOk
          showTodayButton
          helperText={errors?.installationSchedule?.message || ""}
          inputRef={register({
            required: "This is a required field",
            validate: {
              validDate: (value) =>
                moment(value, "MM/DD/YYYY", true).isValid() || "Invalid date",
            },
          })}
          error={!!errors.installationSchedule}
          value={installationSchedule}
          onChange={handleInstalationChange}
          KeyboardButtonProps={{
            "aria-label": "change date",
          }}
          inputProps={{
            className: classes.textField,
          }}
          inputVariant="outlined"
        />
      </MuiPickersUtilsProvider>

      <Typography>
        Installation, survey and maintenance personnel and equipment
      </Typography>
      <TextField
        className={`${classes.formField} ${classes.textField}`}
        variant="outlined"
        inputProps={{ className: classes.textField }}
        fullWidth
        multiline
        rows={4}
        defaultValue={application?.installationResources || null}
        placeholder="Please provide a description of the people that will be able to conduct periodic surveys and maintenance of the buoy. Please also include a description of the equipment (e.g. a boat, cameras) that are available."
        name="installationResources"
        inputRef={register({
          required: "This is a required field",
        })}
        error={!!errors.installationResources}
        helperText={errors?.installationResources?.message || ""}
      />

      <Button
        disabled={!agreed}
        type="submit"
        color="primary"
        variant="contained"
      >
        Submit
      </Button>
    </form>
  );
};

const styles = (theme: Theme) =>
  createStyles({
    form: {
      marginBottom: "3rem",
    },
    formTitle: {
      marginBottom: "2rem",
    },
    formField: {
      marginBottom: "3rem",
    },
    additionalInfo: {
      marginBottom: "3rem",
    },
    scheduleDescription: {
      fontWeight: 300,
      marginBottom: "0.5rem",
    },
    textField: {
      color: "black",
      "&:hover .MuiOutlinedInput-root .MuiOutlinedInput-notchedOutline": {
        borderColor: "rgba(0, 0, 0, 0.23)",
      },
      "& .MuiOutlinedInput-root.Mui-focused .MuiOutlinedInput-notchedOutline": {
        borderColor: theme.palette.primary.main,
      },
    },
  });

interface FormIncomingProps {
  reefName?: string | null;
  application?: ReefApplication | null;
  agreed: boolean;
  handleFormSubmit: (siteName: string, params: ReefApplyParams) => void;
}

Form.defaultProps = {
  reefName: null,
  application: null,
};

type FormProps = FormIncomingProps & WithStyles<typeof styles>;

export default withStyles(styles)(Form);
