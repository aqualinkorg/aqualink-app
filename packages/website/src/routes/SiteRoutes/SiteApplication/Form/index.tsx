import React, { useCallback, useState, useEffect } from 'react';
import { Theme, Typography, TextField, Button } from '@mui/material';
import { WithStyles } from '@mui/styles';
import withStyles from '@mui/styles/withStyles';
import createStyles from '@mui/styles/createStyles';
import { useForm, Controller } from 'react-hook-form';
import { SiteApplication, SiteApplyParams } from 'store/Sites/types';
import { DateTime } from 'luxon-extensions';
import { DatePicker, LocalizationProvider } from '@mui/x-date-pickers';
import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns';

interface SiteApplicationFormFields {
  siteName: string;
  permitRequirements: string;
  fundingSource: string;
  installationResources: string;
  installationSchedule: string;
}

const Form = ({
  siteName,
  application = null,
  agreed,
  handleFormSubmit,
  classes,
}: FormProps) => {
  const [installationSchedule, setInstallationSchedule] = useState<Date | null>(
    null,
  );

  useEffect(() => {
    if (application?.installationSchedule) {
      setInstallationSchedule(new Date(application.installationSchedule));
    }
  }, [application]);

  const {
    formState: { errors },
    handleSubmit,
    setValue,
    control,
  } = useForm<SiteApplicationFormFields>({
    reValidateMode: 'onSubmit',
  });

  const handleInstallationChange = (date: Date | null) => {
    if (date) {
      setInstallationSchedule(date);
    }
  };

  const formSubmit = useCallback(
    (data: SiteApplicationFormFields) => {
      const params: SiteApplyParams = {
        fundingSource: data.fundingSource,
        permitRequirements: data.permitRequirements,
        installationResources: data.installationResources,
        installationSchedule: new Date(data.installationSchedule).toISOString(),
      };
      handleFormSubmit(data.siteName, params);
    },
    [handleFormSubmit],
  );

  return (
    <form className={classes.form} onSubmit={handleSubmit(formSubmit)}>
      <Typography className={classes.formTitle} variant="h3">
        Your Site
      </Typography>
      <TextField
        className={`${classes.formField} ${classes.textField}`}
        variant="outlined"
        fullWidth
        placeholder="Site Name e.g. 'Sombrero Site'"
        disabled
        value={siteName}
        error={!!errors.siteName}
        helperText={errors?.siteName?.message || ''}
        slotProps={{
          htmlInput: { className: classes.textField },
        }}
      />
      <Typography className={classes.additionalInfo}>
        Please provide some additional information for each site:
      </Typography>
      <Typography>Permitting</Typography>
      <Controller
        name="permitRequirements"
        control={control}
        rules={{
          required: 'This is a required field',
        }}
        render={({ field }) => (
          <TextField
            {...field}
            className={`${classes.formField} ${classes.textField}`}
            variant="outlined"
            fullWidth
            multiline
            minRows={2}
            defaultValue={application?.permitRequirements || null}
            placeholder="Please describe the permitting requirements. Please be sure to mention the authority having jurisdiction."
            error={!!errors.permitRequirements}
            helperText={errors?.permitRequirements?.message || ''}
            slotProps={{
              htmlInput: { className: classes.textField },
            }}
          />
        )}
      />
      <Typography>Funding Source</Typography>
      <Controller
        name="fundingSource"
        control={control}
        rules={{
          required: 'This is a required field',
        }}
        render={({ field }) => (
          <TextField
            {...field}
            className={`${classes.formField} ${classes.textField}`}
            variant="outlined"
            fullWidth
            multiline
            minRows={2}
            defaultValue={application?.fundingSource || null}
            placeholder="Funding source for import duties and shipping. Please describe the funding source for the import duties and shipping costs."
            error={!!errors.fundingSource}
            helperText={errors?.fundingSource?.message || ''}
            slotProps={{
              htmlInput: { className: classes.textField },
            }}
          />
        )}
      />
      <Typography>Schedule for installation</Typography>
      <Typography className={classes.scheduleDescription}>
        What is the soonest date you could install the Smart Buoy and conduct a
        survey?
      </Typography>
      <LocalizationProvider dateAdapter={AdapterDateFns}>
        <Controller
          name="installationSchedule"
          control={control}
          rules={{
            required: 'This is a required field',
            validate: {
              validDate: (value) =>
                DateTime.fromFormat(value, 'LL/dd/yyyy').isValid ||
                'Invalid date',
            },
          }}
          render={({ field }) => (
            <DatePicker
              ref={field.ref}
              className={classes.formField}
              format="MM/dd/yyyy"
              closeOnSelect
              value={installationSchedule}
              onChange={(e) => {
                field.onChange(e);
                setValue(
                  'installationSchedule',
                  DateTime.fromJSDate(e || new Date(NaN)).toFormat(
                    'LL/dd/yyyy',
                  ),
                );
                handleInstallationChange(e);
              }}
              slotProps={{
                toolbar: {
                  hidden: true,
                },

                textField: {
                  variant: 'outlined',
                  fullWidth: true,
                  error: !!errors.installationSchedule,
                  helperText: errors?.installationSchedule?.message || '',
                  className: classes.textField,
                },

                openPickerButton: {
                  'aria-label': 'change date',
                },
              }}
            />
          )}
        />
      </LocalizationProvider>
      <Typography>
        Installation, survey and maintenance personnel and equipment
      </Typography>
      <Controller
        name="installationResources"
        control={control}
        rules={{
          required: 'This is a required field',
        }}
        render={({ field }) => (
          <TextField
            {...field}
            className={`${classes.formField} ${classes.textField}`}
            variant="outlined"
            fullWidth
            multiline
            minRows={4}
            defaultValue={application?.installationResources || null}
            placeholder="Please provide a description of the people that will be able to conduct periodic surveys and maintenance of the buoy. Please also include a description of the equipment (e.g. a boat, cameras) that are available."
            error={!!errors.installationResources}
            helperText={errors?.installationResources?.message || ''}
            slotProps={{
              htmlInput: { className: classes.textField },
            }}
          />
        )}
      />
      <Button
        disabled={!agreed}
        type="submit"
        color="primary"
        variant="contained"
      >
        Submit
      </Button>
    </form>
  );
};

const styles = (theme: Theme) =>
  createStyles({
    form: {
      marginBottom: '3rem',
    },
    formTitle: {
      marginBottom: '2rem',
    },
    formField: {
      marginBottom: '3rem',
    },
    additionalInfo: {
      marginBottom: '3rem',
    },
    scheduleDescription: {
      fontWeight: 300,
      marginBottom: '0.5rem',
    },
    textField: {
      color: 'black',
      '&:hover .MuiOutlinedInput-root .MuiOutlinedInput-notchedOutline': {
        borderColor: 'rgba(0, 0, 0, 0.23)',
      },
      '& .MuiOutlinedInput-root.Mui-focused .MuiOutlinedInput-notchedOutline': {
        borderColor: theme.palette.primary.main,
      },
    },
  });

interface FormIncomingProps {
  siteName: string;
  application?: SiteApplication | null;
  agreed: boolean;
  handleFormSubmit: (siteName: string, params: SiteApplyParams) => void;
}

type FormProps = FormIncomingProps & WithStyles<typeof styles>;

export default withStyles(styles)(Form);
