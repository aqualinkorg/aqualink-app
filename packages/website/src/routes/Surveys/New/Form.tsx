import React, { useCallback } from "react";
import { Grid, Collapse, IconButton } from "@material-ui/core";
import Alert from "@material-ui/lab/Alert";
import { useSelector, useDispatch } from "react-redux";
import { SurveyData, SurveyState } from "../../../store/Survey/types";
import {
  surveyErrorSelector,
  surveyAddRequest,
} from "../../../store/Survey/surveySlice";
import { userInfoSelector } from "../../../store/User/userSlice";
import Form from "../../../common/SurveyForm";

const SurveyForm = ({ reefId, timeZone, changeTab }: SurveyFormProps) => {
  const user = useSelector(userInfoSelector);
  const surveyError = useSelector(surveyErrorSelector);

  const dispatch = useDispatch();

  const onSubmit = useCallback(
    (
      diveDateTime: string,
      diveLocation: SurveyState["diveLocation"],
      weatherConditions: SurveyData["weatherConditions"],
      comments: string
    ) => {
      const surveyData: SurveyData = {
        reef: reefId,
        diveDate: diveDateTime,
        diveLocation,
        weatherConditions,
        comments,
        token: user?.token,
      };
      dispatch(
        surveyAddRequest({
          reefId: `${reefId}`,
          surveyData,
          changeTab,
        })
      );
    },
    [dispatch, changeTab, reefId, user]
  );

  return (
    <>
      <Grid item xs={12}>
        <Collapse in={!!surveyError}>
          <Alert
            severity="error"
            action={
              <IconButton aria-label="close" color="inherit" size="small" />
            }
          >
            There was an error creating the survey.
          </Alert>
        </Collapse>
      </Grid>
      <Form reefId={reefId} timeZone={timeZone} onSubmit={onSubmit} />
    </>
  );
};

interface SurveyFormProps {
  changeTab: (index: number) => void;
  reefId: number;
  timeZone?: string | null;
}

SurveyForm.defaultProps = {
  timeZone: null,
};

export default SurveyForm;
