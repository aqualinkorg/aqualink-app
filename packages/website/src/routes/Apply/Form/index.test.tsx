import React from "react";
import { render } from "@testing-library/react";

import Form from ".";

jest.mock("@material-ui/pickers", () => ({
  MuiPickersUtilsProvider: () => "Mock-MuiPickersUtilsProvider",
}));

test("renders as expected", () => {
  const { container } = render(
    <Form reefName="Mock Reef" agreed handleFormSubmit={jest.fn()} />
  );
  expect(container).toMatchSnapshot();
});
