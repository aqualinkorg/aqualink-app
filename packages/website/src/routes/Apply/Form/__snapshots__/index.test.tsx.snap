// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div>
  <form
    class="Form-form-1"
  >
    <mock-typography
      classname="Form-formTitle-2"
      variant="h3"
    >
      Your Site
    </mock-typography>
    <mock-textfield
      classname="Form-formField-3 Form-textField-6"
      disabled="true"
      error="false"
      fullwidth="true"
      helpertext=""
      inputprops="[object Object]"
      name="siteName"
      variant="outlined"
    />
    <mock-typography
      classname="Form-additionalInfo-4"
    >
      Please provide some additional information for each reef:
    </mock-typography>
    <mock-typography>
      Permitting
    </mock-typography>
    <mock-textfield
      classname="Form-formField-3 Form-textField-6"
      error="false"
      fullwidth="true"
      helpertext=""
      inputprops="[object Object]"
      multiline="true"
      name="permitRequirements"
      placeholder="Please describe the permitting requirements. Please be sure to mention the authority having jurisdiction"
      rows="2"
      variant="outlined"
    />
    <mock-typography>
      Funding Source
    </mock-typography>
    <mock-textfield
      classname="Form-formField-3 Form-textField-6"
      error="false"
      fullwidth="true"
      helpertext=""
      inputprops="[object Object]"
      multiline="true"
      name="fundingSource"
      placeholder="Funding source for import duties and shipping. Please describe the funding source for the import duties and shipping costs"
      rows="2"
      variant="outlined"
    />
    <mock-typography>
      Schedule for installation
    </mock-typography>
    <mock-typography
      classname="Form-scheduleDescription-5"
    >
      What is the soonest date you could install the spotter and conduct a survey
    </mock-typography>
    <mock-textfield
      classname="Form-formField-3"
      error="false"
      helpertext=""
      id="installationSchedule"
      inputprops="[object Object]"
      name="installationSchedule"
      value=""
      variant="outlined"
    />
    <mock-dialog
      classes="[object Object]"
      open="false"
      role="dialog"
    >
      <mock-dialogcontent
        classname="MuiPickersModal-dialog"
      >
        <div
          class="MuiPickersBasePicker-container"
        >
          <div
            class="MuiPickersBasePicker-pickerView"
          >
            <div>
              <div
                class="MuiPickersCalendarHeader-switchHeader"
              >
                <mock-iconbutton
                  classname="MuiPickersCalendarHeader-iconButton"
                  disabled="false"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
                    />
                    <path
                      d="M0 0h24v24H0V0z"
                      fill="none"
                    />
                  </svg>
                </mock-iconbutton>
                <div
                  class="MuiPickersSlideTransition-transitionContainer MuiPickersCalendarHeader-transitionContainer"
                >
                  <mock-typography
                    align="center"
                    variant="body1"
                  >
                    October 2020
                  </mock-typography>
                </div>
                <mock-iconbutton
                  classname="MuiPickersCalendarHeader-iconButton"
                  disabled="false"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
                    />
                    <path
                      d="M0 0h24v24H0V0z"
                      fill="none"
                    />
                  </svg>
                </mock-iconbutton>
              </div>
              <div
                class="MuiPickersCalendarHeader-daysHeader"
              >
                <mock-typography
                  classname="MuiPickersCalendarHeader-dayLabel"
                  variant="caption"
                >
                  Su
                </mock-typography>
                <mock-typography
                  classname="MuiPickersCalendarHeader-dayLabel"
                  variant="caption"
                >
                  Mo
                </mock-typography>
                <mock-typography
                  classname="MuiPickersCalendarHeader-dayLabel"
                  variant="caption"
                >
                  Tu
                </mock-typography>
                <mock-typography
                  classname="MuiPickersCalendarHeader-dayLabel"
                  variant="caption"
                >
                  We
                </mock-typography>
                <mock-typography
                  classname="MuiPickersCalendarHeader-dayLabel"
                  variant="caption"
                >
                  Th
                </mock-typography>
                <mock-typography
                  classname="MuiPickersCalendarHeader-dayLabel"
                  variant="caption"
                >
                  Fr
                </mock-typography>
                <mock-typography
                  classname="MuiPickersCalendarHeader-dayLabel"
                  variant="caption"
                >
                  Sa
                </mock-typography>
              </div>
            </div>
            <div
              class="MuiPickersSlideTransition-transitionContainer MuiPickersCalendar-transitionContainer"
            >
              <div>
                <div
                  class="MuiPickersCalendar-week"
                >
                  <div
                    role="presentation"
                  >
                    <mock-iconbutton
                      classname="MuiPickersDay-day MuiPickersDay-hidden"
                      tabindex="-1"
                    >
                      <mock-typography
                        color="inherit"
                        variant="body2"
                      >
                        27
                      </mock-typography>
                    </mock-iconbutton>
                  </div>
                  <div
                    role="presentation"
                  >
                    <mock-iconbutton
                      classname="MuiPickersDay-day MuiPickersDay-hidden"
                      tabindex="-1"
                    >
                      <mock-typography
                        color="inherit"
                        variant="body2"
                      >
                        28
                      </mock-typography>
                    </mock-iconbutton>
                  </div>
                  <div
                    role="presentation"
                  >
                    <mock-iconbutton
                      classname="MuiPickersDay-day MuiPickersDay-hidden"
                      tabindex="-1"
                    >
                      <mock-typography
                        color="inherit"
                        variant="body2"
                      >
                        29
                      </mock-typography>
                    </mock-iconbutton>
                  </div>
                  <div
                    role="presentation"
                  >
                    <mock-iconbutton
                      classname="MuiPickersDay-day MuiPickersDay-hidden"
                      tabindex="-1"
                    >
                      <mock-typography
                        color="inherit"
                        variant="body2"
                      >
                        30
                      </mock-typography>
                    </mock-iconbutton>
                  </div>
                  <div
                    role="presentation"
                  >
                    <mock-iconbutton
                      classname="MuiPickersDay-day"
                      tabindex="0"
                    >
                      <mock-typography
                        color="inherit"
                        variant="body2"
                      >
                        1
                      </mock-typography>
                    </mock-iconbutton>
                  </div>
                  <div
                    role="presentation"
                  >
                    <mock-iconbutton
                      classname="MuiPickersDay-day"
                      tabindex="0"
                    >
                      <mock-typography
                        color="inherit"
                        variant="body2"
                      >
                        2
                      </mock-typography>
                    </mock-iconbutton>
                  </div>
                  <div
                    role="presentation"
                  >
                    <mock-iconbutton
                      classname="MuiPickersDay-day"
                      tabindex="0"
                    >
                      <mock-typography
                        color="inherit"
                        variant="body2"
                      >
                        3
                      </mock-typography>
                    </mock-iconbutton>
                  </div>
                </div>
                <div
                  class="MuiPickersCalendar-week"
                >
                  <div
                    role="presentation"
                  >
                    <mock-iconbutton
                      classname="MuiPickersDay-day"
                      tabindex="0"
                    >
                      <mock-typography
                        color="inherit"
                        variant="body2"
                      >
                        4
                      </mock-typography>
                    </mock-iconbutton>
                  </div>
                  <div
                    role="presentation"
                  >
                    <mock-iconbutton
                      classname="MuiPickersDay-day"
                      tabindex="0"
                    >
                      <mock-typography
                        color="inherit"
                        variant="body2"
                      >
                        5
                      </mock-typography>
                    </mock-iconbutton>
                  </div>
                  <div
                    role="presentation"
                  >
                    <mock-iconbutton
                      classname="MuiPickersDay-day"
                      tabindex="0"
                    >
                      <mock-typography
                        color="inherit"
                        variant="body2"
                      >
                        6
                      </mock-typography>
                    </mock-iconbutton>
                  </div>
                  <div
                    role="presentation"
                  >
                    <mock-iconbutton
                      classname="MuiPickersDay-day"
                      tabindex="0"
                    >
                      <mock-typography
                        color="inherit"
                        variant="body2"
                      >
                        7
                      </mock-typography>
                    </mock-iconbutton>
                  </div>
                  <div
                    role="presentation"
                  >
                    <mock-iconbutton
                      classname="MuiPickersDay-day"
                      tabindex="0"
                    >
                      <mock-typography
                        color="inherit"
                        variant="body2"
                      >
                        8
                      </mock-typography>
                    </mock-iconbutton>
                  </div>
                  <div
                    role="presentation"
                  >
                    <mock-iconbutton
                      classname="MuiPickersDay-day"
                      tabindex="0"
                    >
                      <mock-typography
                        color="inherit"
                        variant="body2"
                      >
                        9
                      </mock-typography>
                    </mock-iconbutton>
                  </div>
                  <div
                    role="presentation"
                  >
                    <mock-iconbutton
                      classname="MuiPickersDay-day"
                      tabindex="0"
                    >
                      <mock-typography
                        color="inherit"
                        variant="body2"
                      >
                        10
                      </mock-typography>
                    </mock-iconbutton>
                  </div>
                </div>
                <div
                  class="MuiPickersCalendar-week"
                >
                  <div
                    role="presentation"
                  >
                    <mock-iconbutton
                      classname="MuiPickersDay-day"
                      tabindex="0"
                    >
                      <mock-typography
                        color="inherit"
                        variant="body2"
                      >
                        11
                      </mock-typography>
                    </mock-iconbutton>
                  </div>
                  <div
                    role="presentation"
                  >
                    <mock-iconbutton
                      classname="MuiPickersDay-day MuiPickersDay-current MuiPickersDay-daySelected"
                      tabindex="0"
                    >
                      <mock-typography
                        color="inherit"
                        variant="body2"
                      >
                        12
                      </mock-typography>
                    </mock-iconbutton>
                  </div>
                  <div
                    role="presentation"
                  >
                    <mock-iconbutton
                      classname="MuiPickersDay-day"
                      tabindex="0"
                    >
                      <mock-typography
                        color="inherit"
                        variant="body2"
                      >
                        13
                      </mock-typography>
                    </mock-iconbutton>
                  </div>
                  <div
                    role="presentation"
                  >
                    <mock-iconbutton
                      classname="MuiPickersDay-day"
                      tabindex="0"
                    >
                      <mock-typography
                        color="inherit"
                        variant="body2"
                      >
                        14
                      </mock-typography>
                    </mock-iconbutton>
                  </div>
                  <div
                    role="presentation"
                  >
                    <mock-iconbutton
                      classname="MuiPickersDay-day"
                      tabindex="0"
                    >
                      <mock-typography
                        color="inherit"
                        variant="body2"
                      >
                        15
                      </mock-typography>
                    </mock-iconbutton>
                  </div>
                  <div
                    role="presentation"
                  >
                    <mock-iconbutton
                      classname="MuiPickersDay-day"
                      tabindex="0"
                    >
                      <mock-typography
                        color="inherit"
                        variant="body2"
                      >
                        16
                      </mock-typography>
                    </mock-iconbutton>
                  </div>
                  <div
                    role="presentation"
                  >
                    <mock-iconbutton
                      classname="MuiPickersDay-day"
                      tabindex="0"
                    >
                      <mock-typography
                        color="inherit"
                        variant="body2"
                      >
                        17
                      </mock-typography>
                    </mock-iconbutton>
                  </div>
                </div>
                <div
                  class="MuiPickersCalendar-week"
                >
                  <div
                    role="presentation"
                  >
                    <mock-iconbutton
                      classname="MuiPickersDay-day"
                      tabindex="0"
                    >
                      <mock-typography
                        color="inherit"
                        variant="body2"
                      >
                        18
                      </mock-typography>
                    </mock-iconbutton>
                  </div>
                  <div
                    role="presentation"
                  >
                    <mock-iconbutton
                      classname="MuiPickersDay-day"
                      tabindex="0"
                    >
                      <mock-typography
                        color="inherit"
                        variant="body2"
                      >
                        19
                      </mock-typography>
                    </mock-iconbutton>
                  </div>
                  <div
                    role="presentation"
                  >
                    <mock-iconbutton
                      classname="MuiPickersDay-day"
                      tabindex="0"
                    >
                      <mock-typography
                        color="inherit"
                        variant="body2"
                      >
                        20
                      </mock-typography>
                    </mock-iconbutton>
                  </div>
                  <div
                    role="presentation"
                  >
                    <mock-iconbutton
                      classname="MuiPickersDay-day"
                      tabindex="0"
                    >
                      <mock-typography
                        color="inherit"
                        variant="body2"
                      >
                        21
                      </mock-typography>
                    </mock-iconbutton>
                  </div>
                  <div
                    role="presentation"
                  >
                    <mock-iconbutton
                      classname="MuiPickersDay-day"
                      tabindex="0"
                    >
                      <mock-typography
                        color="inherit"
                        variant="body2"
                      >
                        22
                      </mock-typography>
                    </mock-iconbutton>
                  </div>
                  <div
                    role="presentation"
                  >
                    <mock-iconbutton
                      classname="MuiPickersDay-day"
                      tabindex="0"
                    >
                      <mock-typography
                        color="inherit"
                        variant="body2"
                      >
                        23
                      </mock-typography>
                    </mock-iconbutton>
                  </div>
                  <div
                    role="presentation"
                  >
                    <mock-iconbutton
                      classname="MuiPickersDay-day"
                      tabindex="0"
                    >
                      <mock-typography
                        color="inherit"
                        variant="body2"
                      >
                        24
                      </mock-typography>
                    </mock-iconbutton>
                  </div>
                </div>
                <div
                  class="MuiPickersCalendar-week"
                >
                  <div
                    role="presentation"
                  >
                    <mock-iconbutton
                      classname="MuiPickersDay-day"
                      tabindex="0"
                    >
                      <mock-typography
                        color="inherit"
                        variant="body2"
                      >
                        25
                      </mock-typography>
                    </mock-iconbutton>
                  </div>
                  <div
                    role="presentation"
                  >
                    <mock-iconbutton
                      classname="MuiPickersDay-day"
                      tabindex="0"
                    >
                      <mock-typography
                        color="inherit"
                        variant="body2"
                      >
                        26
                      </mock-typography>
                    </mock-iconbutton>
                  </div>
                  <div
                    role="presentation"
                  >
                    <mock-iconbutton
                      classname="MuiPickersDay-day"
                      tabindex="0"
                    >
                      <mock-typography
                        color="inherit"
                        variant="body2"
                      >
                        27
                      </mock-typography>
                    </mock-iconbutton>
                  </div>
                  <div
                    role="presentation"
                  >
                    <mock-iconbutton
                      classname="MuiPickersDay-day"
                      tabindex="0"
                    >
                      <mock-typography
                        color="inherit"
                        variant="body2"
                      >
                        28
                      </mock-typography>
                    </mock-iconbutton>
                  </div>
                  <div
                    role="presentation"
                  >
                    <mock-iconbutton
                      classname="MuiPickersDay-day"
                      tabindex="0"
                    >
                      <mock-typography
                        color="inherit"
                        variant="body2"
                      >
                        29
                      </mock-typography>
                    </mock-iconbutton>
                  </div>
                  <div
                    role="presentation"
                  >
                    <mock-iconbutton
                      classname="MuiPickersDay-day"
                      tabindex="0"
                    >
                      <mock-typography
                        color="inherit"
                        variant="body2"
                      >
                        30
                      </mock-typography>
                    </mock-iconbutton>
                  </div>
                  <div
                    role="presentation"
                  >
                    <mock-iconbutton
                      classname="MuiPickersDay-day"
                      tabindex="0"
                    >
                      <mock-typography
                        color="inherit"
                        variant="body2"
                      >
                        31
                      </mock-typography>
                    </mock-iconbutton>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </mock-dialogcontent>
      <mock-dialogactions
        classes="[object Object]"
      >
        <mock-button
          color="primary"
        >
          Today
        </mock-button>
        <mock-button
          color="primary"
        >
          Cancel
        </mock-button>
        <mock-button
          color="primary"
        >
          OK
        </mock-button>
      </mock-dialogactions>
    </mock-dialog>
    <mock-typography>
      Installation, survey and maintenance personnel and equipment
    </mock-typography>
    <mock-textfield
      classname="Form-formField-3 Form-textField-6"
      error="false"
      fullwidth="true"
      helpertext=""
      inputprops="[object Object]"
      multiline="true"
      name="installationResources"
      placeholder="Please provide a description of the people that will be able to conduct periodic surveys and maintenance of the buoy. Please also include a description of the equipment (e.g. a boat, cameras) that are available."
      rows="4"
      variant="outlined"
    />
    <mock-button
      color="primary"
      disabled="false"
      type="submit"
      variant="contained"
    >
      Submit
    </mock-button>
  </form>
</div>
`;
